op-challenger/game/fault/types/position.go

```diff
@@ -25,13 +25,27 @@ func NewPosition(depth int, indexAtDepth *big.Int) Position {
 	}
 }
 
+// NewPositionFromGIndex creates a new Position given a generalized index,
+// numbered like the following:
+//
+//			 1
+//		    /  \
+//	      2     3
+//		 / \   / \
+//		4   5 6   7
+//
+// See ../../../../specs/fault-dispute-game.md#game-tree
 func NewPositionFromGIndex(x *big.Int) Position {
 	depth := bigMSB(x)
 	withoutMSB := new(big.Int).Not(new(big.Int).Lsh(big.NewInt(1), uint(depth)))
 	indexAtDepth := new(big.Int).And(x, withoutMSB)
 	return NewPosition(depth, indexAtDepth)
 }
 
+func (p Position) Equal(other Position) bool {
+	return p.Depth() == other.Depth() && p.IndexAtDepth().Cmp(other.IndexAtDepth()) == 0
+}
```

Comment:
We don't seem to have an actual need for this method.

Reply:
We otherwise expect the caller to know [how to test equality](https://github.com/ethereum-optimism/optimism/blob/0448eb7988bb2247451280f8d6bdc88b9f90cfe5/op-challenger/game/fault/types/position_test.go#L95-L96), but I see that as a violation of encapsulation. The caller doesn't necessarily know all of `Position`'s fields to compare two objects for equality.

One way this can go wrong, if we end up adding a field to `Position` that affects whether two `Position`s are equal, it would be easy to forget to update callers that are testing equality by manually checking its two existing fields, whereas if there's an `Equal` method, that's harder to miss.

Reply:
Which caller needs to test equality? Nothing is actually calling `Equals` except the one test that checks `Equals` is implemented correctly.

Reply:
I see that as more verifying that `NewPositionFromGIndex` is implemented correctly, but it sounds like that use-case isn't compelling, so I've removed it in 96a801b1b48326998d9baf525ac2ebe433b43fa3.